src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -125,8 +129,14 @@ public bool DiscoverTests(DiscoveryRequestPayload discoveryPayload, ITestDiscove
 
             bool success = false;
 
+            var runsettings = discoveryPayload.RunSettings;
+            if (this.TryUpdateDesignMode(runsettings, out string updatedRunsettings))
```

Comment:
Update design mode should be done before any logger initialize in case any logger required designMode prop. This requirement should meet in master branch.